apps/nodeApp/src/androidMain/kotlin/network/bisq/mobile/android/node/service/network/NodeConnectivityService.kt

```diff
@@ -16,14 +20,32 @@ class NodeConnectivityService(
     override fun activate() {
         collectJob?.cancel()
         collectJob = serviceScope.launch {
+            // Load persisted flag before collecting status changes, with version gating
+            val settings = try {
+                settingsRepository.fetch()
+            } catch (e: Exception) {
+                null
+            }
+            val isFirstRunAfterUpgrade = settings?.lastSeenNodeAppVersion != BuildNodeConfig.APP_VERSION
```

Comment:
We do you check for the version? The everReceivedAllData is not really relevant to an app version update.